digdag-server/src/main/java/io/digdag/server/WorkflowExecutionTimeoutEnforcer.java

```diff
@@ -46,11 +46,14 @@
     private static final Duration DEFAULT_REAPING_INTERVAL = Duration.ofSeconds(5);
 
     // This is similar to TaskStateCode.notDoneStates() but BLOCKED, PLANNED, and READY are excluded.
-    // BLOCKED and PLANNED are excluded because there're other running tasks that should be enforced instead.
-    // READY is excluded the workflow itself is working correctly and number of threads is insufficient.
+    //
+    // - BLOCKED and PLANNED are excluded because there're other running tasks that should be enforced instead.
+    // - READY is excluded the workflow itself is working correctly and number of threads is insufficient.
+    // - GROUP_RETRY_WAITING is excluded because this status is supposed to be immediately changed to PLANNED
+    //   and long running group tasks with this state could happen only due to WorkflowExecutor trouble
+    //   not due to workflow definitions.
     private static final TaskStateCode[] TASK_TTL_ENFORCED_STATE_CODES = new TaskStateCode[] {
         TaskStateCode.RETRY_WAITING,
-        TaskStateCode.GROUP_RETRY_WAITING,
```

Comment:
GROUP_RETRY_WAITING should be excluded in this change since if a group task has sequential 30 children tasks which takes 1 hour and the group task is in GROUP_RETRY_WAITING state, the group task will be regarded as an expired task.

Excluding this state should be okay because in current implementation group tasks don't have `started_at` and they're ignored by `WorkflowExecutionTimeoutEnforcer`. In a sense this state in this list hasn't made sense.

Also, non-group task won't be this state.

https://github.com/treasure-data/digdag/blob/432517d0141959a7bc906790ce99233bc39544ce/digdag-core/src/main/java/io/digdag/core/workflow/WorkflowExecutor.java#L712-L722

https://github.com/treasure-data/digdag/blob/432517d0141959a7bc906790ce99233bc39544ce/digdag-core/src/main/java/io/digdag/core/database/DatabaseSessionStoreManager.java#L1036-L1045